### PR TITLE
Prevents recursion of arrays

### PR DESCRIPTION
**Pull request by melonmanchan**
#24 

Is there a smarter/more effective way of doing this? Feedback is much appreciated!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,10 +25,19 @@ module.exports = function (val, opts, pad) {
...
+
+			if (seen.indexOf(val) !== -1) {
+				return '"[Circular]"';
+			}
+
 			if (val.length === 0) {
 				return '[]';
 			}
 
+			if (arrayEqual(val, val)) {
```
Comment (danilosampaio) [inline index.js]: @melonmanchan what the purpose of comparing `val` with itself?

**Comment by melonmanchan** [comment]
Ah, you're right. Simply comparing the array to itself leads to false positives if the array simply happens to contain another array with the same elements, which doesn't necessarily mean it contains itself. This still needs work!

File: index.js [hunk 2]
```diff
@@ -24,10 +24,17 @@ module.exports = function (val, opts, pad) {
 		}
 
 		if (Array.isArray(val)) {
+
```
Comment (danilosampaio) [inline index.js]: @melonmanchan Unnecessary new line.
You may want to install `xo` to help you with the existing codestyle: `npm install xo`

**Comment by sindresorhus** [comment]
This has false positives.

```js
var val = [10];
stringifyObject({foo: val, bar: val});
/*
{
	foo: [
		10
	],
	bar: "[Circular]"
}
*/
```

This is just a reused value, not a circular reference.

**Comment by melonmanchan** [comment]
Sorry, had completely forgotten about this PR until now. Added tests for and resolved the false positive issue!

**Comment by sindresorhus** [comment]
I don't think this is the correct solution.

Quickly found a way to break it:

```js
it('should handle circular recursion in arrays', function () {
	var array = [[]];
	array[0][0] = array;

	assert.doesNotThrow(
		function () {
			stringifyObject(array);
		}, RangeError);
});
```

**Comment by melonmanchan** [comment]
OK, I'll still keep at it. Thanks for your patience!

**Comment by melonmanchan** [comment]
Had a go at it once again, how's it look?

**Comment by sindresorhus** [comment]
```js
it('should handle circular recursion in arrays', function () {
    var array2 = [];
    var array = [array2];
    array2[0] = array2;

    assert.doesNotThrow(
        function () {
            stringifyObject(array);
        }, RangeError);
});
//=> AssertionError: Got unwanted exception (RangeError)..
```

**Comment by sindresorhus** [comment]
I've merged your failing tests, but I the fix is not correct and I don't have time for more back and forth ;)

**Comment by melonmanchan** [comment]
Fair enough, I'll leave it to someone smarter than me!

**Pull request closed without merging**